src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace D2L.CodeStyle.Analyzers.Async.Generator;
+
+internal sealed class AsyncToSyncMethodTransformer : SyntaxTransformer {

```

Comment:
This file is where the fun is going to happen. It will be a lot bigger ultimately (as it will need to handle every kind of syntax) but from my prototyping I think it can be really dumb/stateless.